### PR TITLE
test: Update real service tests to output test failures to the console

### PR DESCRIPTION
**Pull request by Josmithr**
Currently, when the "real service" tests run in CI, nothing is reported to the console when tests fail. The only way to differentiate infra failures from test run failures is to look at the published test results artifact. This PR attempts to improve that by adding test failure logging to the console output during the run.

Utilizes infrastructure added in #14702 to ensure that failures in test runs are logged to console output in real service tests. This should make it more immediately clear when a pipeline was unsuccessful due to failures during the test run rather than some other infrastructural issue.

### CODE REVIEW

File: packages/test/test-version-utils/src/compatConfig.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ const genFullBackCompatConfig = (): CompatConfig[] => {
 	} else {
 		// This will need to be updated once we move beyond 2.0.0-internal.x.y.z
 		throw new Error(
-			"Unexpected back compat scenario! Expecting package versiosn to just be 2.0.0-internal.x.y.z",
+			`Unexpected back compat scenario! Expecting package versions to just be 2.0.0-internal.x.y.z, but got ${pkgVersion}.`,
```
Comment (Josmithr) [inline packages/test/test-version-utils/src/compatConfig.ts:164]: Ran into this while testing via a manual run of the pipeline on CI. Logging the version found was useful.

**Comment by Josmithr** [comment]
Sample successful pipeline run (via build: client): https://dev.azure.com/fluidframework/internal/_build/results?buildId=167425&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=c32d47e1-65cf-544f-3c15-66e3d2029546

Sample unsuccessful run with the addition of a test that unconditionally failed: https://dev.azure.com/fluidframework/internal/_build/results?buildId=167487&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=ca111d9b-56f7-5b3d-b8f6-661164061440

Note that the failure case is not logging the particular tests that failed, but it is logging information indicating that there was a single test failure that occurred during the run (which is not data we saw previously).

File: packages/test/test-end-to-end-tests/package.json [hunk 2]
```diff
@@ -134,6 +134,9 @@
 		"@types/uuid": "^8.3.0",
 		"concurrently": "^7.6.0",
 		"eslint": "~8.6.0",
+		"mocha-json-output-reporter": "^2.0.1",
+		"mocha-multi-reporters": "^1.5.1",
+		"moment": "^2.21.0",
```
Comment (curtisman) [inline packages/test/test-end-to-end-tests/package.json:139]: I know this is in the original change, but I wonder why this is needed.

**Review by curtisman — APPROVED** [review]

**Review by alexvy86 — APPROVED** [review]
Same question as Curtis; `moment` seems like it shouldn't be necessary.

**Review by scottn12 — APPROVED** [review]

File: packages/test/test-end-to-end-tests/package.json (hunk 2 above)
Comment (Josmithr) [reply packages/test/test-end-to-end-tests/package.json:139]: I was under the impression that it was a peer of one of the other two, but I will confirm.

Comment (Josmithr) [reply packages/test/test-end-to-end-tests/package.json:139]: Yes, it is a peer dep of `mocha-json-output-reporter`.

**Pull request merged**